network/src/peer_store/addr_manager.rs

```diff
@@ -1,13 +1,14 @@
 //! Address manager
-use crate::peer_store::types::{AddrInfo, IpPort};
+use crate::peer_store::types::AddrInfo;
+use p2p::{multiaddr::Multiaddr, utils::multiaddr_to_socketaddr};
 use rand::Rng;
 use std::collections::{HashMap, HashSet};
 
 /// Address manager
 #[derive(Default)]
 pub struct AddrManager {
     next_id: u64,
-    addr_to_id: HashMap<IpPort, u64>,
+    addr_to_id: HashMap<Multiaddr, u64>,
```

Comment:
We can append any protocol to the Multiaddr format in any order.

For example `ip/127.0.0.1/port/443/ws/127.0.0.2` is identical to `port/443/ip/127.0.0.1` in our system, that's the reason why we **can't use the string as key** here.

Reply:
I don't understand the reason, could you explain more detail?

Reply:
Consider the scenario:

1. We connected with address: `ip/127.0.0.1/port/443`.
2. Then we receive a new address `port/443/ip/127.0.0.1`.
3. The code after refactoring will determine we have two different addresses, it may cause inconsistent behavior.

Reply:
understood, I've updated the code, please review again, thanks.

Reply:
`port/443/ip/127.0.0.1` will be rejected by Tentacle, although it is a legal multiaddr